src/GithubPlugin.fs

```diff
@@ -127,27 +136,42 @@ type GithubPlugin() =
         | _ ->
             [ defaultResult ]
 
-    let tryRunApiSearch =
-           Async.Catch
-        >> Async.RunSynchronously
-        >> function
-            | Choice1Of2 result -> presentApiSearchResult result
-            | Choice2Of2 exn -> presentApiSearchExn exn
+    let tryRunApiSearch (result: Async<ApiSearchResult>) =
+        async {
+            let! res = result |> Async.Catch
 
+            return
+                match res with
+                | Choice1Of2 result -> presentApiSearchResult result
+                | Choice2Of2 exn -> presentApiSearchExn exn
+        }
     member this.ProcessQuery terms =
-        match parseQuery terms with
-        | RunApiSearch fSearch -> tryRunApiSearch fSearch
-        | SuggestQuery suggestion -> presentSuggestion suggestion
-
-    interface IPlugin with
-        member this.Init (context:PluginInitContext) =
+        async {
+            let! res =
+                match parseQuery terms with
+                | RunApiSearch fSearch -> tryRunApiSearch fSearch
+                | SuggestQuery suggestion -> async { return presentSuggestion suggestion }
+
+            return
+                res
+                |> List.map
+                    (fun r ->
+                        Result(
+                            Title = r.title,
+                            SubTitle = r.subtitle,
+                            IcoPath = "icon.png",
+                            Action = fun x -> r.action x
+                        ))
+                |> List<Result>
+        }
+
+    interface IAsyncPlugin with
+        member this.InitAsync(context: PluginInitContext) =
             Helpers.githubTokenFileDir <- context.CurrentPluginMetadata.PluginDirectory
+
             pluginContext <- context
+            Task.CompletedTask
+
 
-        member this.Query (query:Query) =
-            query.Terms
-            |> List.ofArray
-            |> List.skip 1
-            |> this.ProcessQuery
-            |> List.map (fun r -> Result( Title = r.title, SubTitle = r.subtitle, IcoPath = "icon.png", Action = fun x -> r.action x ))
-            |> List<Result>
+        member this.QueryAsync(query, token: CancellationToken) =
+            Async.StartImmediateAsTask(this.ProcessQuery query, token)
```

Comment:
Continuing from above, you can declare a local async function here that passes `query.Terms` down to `ProcessQuery` and then maps the results to a `List<Flow.Launcher.Plugin.Result>`, and then pass that function to `Async.StartImmediateAsTask`.

Reply:
So to clarify, the local function could look something like:

```fsharp
let f = async {
    let! results = 
        query.Terms
        |> List.ofArray
        |> List.skip 1
        |> this.ProcessQuery
    
    return results
           |> List.map (fun r -> Result( Title = r.title, SubTitle = r.subtitle, IcoPath = "icon.png", Action = fun x -> r.action x ))
           |> List<Result>
}
```